i2ccontroller/I2CController.cpp

```diff
@@ -1,95 +1,94 @@
+		return returnVector;
+	}
 
-    initCompass(m_mockCompass, m_headingBufferSize);
-    initArduino(m_mockArduino);
+	return {}; //This should never happen unless you are silly. beginTransmission needs to be called before this function
 }
 
-void I2CController::run() {
-    Logger::info("I2CController thread started");
-    while(isRunning()) {
-        m_timer.reset();
-
-        m_compass->readValues();
-        m_compass->readAccel();
-
-        //update system state
-        m_systemState->setCompassModel(CompassModel(
-        m_compass->getHeading(),
-        m_compass->getPitch(),
-        m_compass->getRoll(),
-        AccelerationModel(m_compass->getAccelX(),
-                          m_compass->getAccelY(),
-                          m_compass->getAccelZ())
-        ));
-        m_arduino->readValues();
-        m_systemState->setAnalogArduinoModel(m_arduino->getModel());
-
-        m_timer.sleepUntil(m_loopTime);
-    }
+int I2CController::write(int data)
+{
+	if (locked)
+	{
+		return wiringPiI2CWrite(m_fd, data);
+	}
+
+	return 0; //This should never happen unless you are silly. beginTransmission needs to be called before this function
 }
 
-bool I2CController::initCompass(bool mockCompass, int headningBufferSize) {
-    if (!mockCompass) {
-        m_compass.reset(new HMC6343(headningBufferSize) );
-    } else {
-        m_compass.reset(new MockCompass());
-    }
-
-    if(not m_compass->init())
-    {
-        Logger::error("%s Failed to initialise the compass", __PRETTY_FUNCTION__);
-        return false;
-    }
-
-    Logger::info("Compass initialised");
-    return true;
+int I2CController::read()
+{
+	if (locked)
+	{
+		return wiringPiI2CRead(m_fd);
+	}
+
+	return 0; //This should never happen unless you are silly. beginTransmission needs to be called before this function
 }
 
-bool I2CController::initArduino(bool mockArduino) {
-    if(!mockArduino) {
-        m_arduino.reset(new AR_UNO());
-    } else {
-        m_arduino.reset(new MockAnalogArduino());
-    }
-
-    if(not m_arduino->init())
-    {
-        Logger::error("%s Failed to communicate with the arduino", __PRETTY_FUNCTION__);
-        return false;
-    }
-
-    Logger::info("Arduino initialised");
-    return true;
+int I2CController::readBlock()
+{
+	if(locked)
+	{
+		uint8_t block[40];
+		wiringPiI2CReadBlock(m_fd, block);
+
+		return block[10];
+	}
+
+	return 0; //This should never happen unless you are silly. beginTransmission needs to be called before this function
 }
 
-bool I2CController::isRunning() {
-    bool running;
-    m_mutex.lock();
-    running = m_running;
-    m_mutex.unlock();
-    return running;
+
+void I2CController::beginTransmission()
+{
+	m_mutex.lock();
+	locked = true;
 }
 
-void I2CController::close() {
-    m_mutex.lock();
-    m_running = false;
-    m_mutex.unlock();
+void I2CController::endTransmission()
+{
+	locked = false;
```

Comment:
This is a void function, so is locked a class variable ? If yes, why it's not named m_locked ?